warehouse/query-core/src/main/java/datawave/query/composite/CompositeTerm.java

```diff
@@ -85,7 +89,21 @@ public void getNodesAndExpressions(List<Class<? extends JexlNode>> nodeClasses,
      * @return
      */
     private String getAppendedExpressions() {
-        return String.join(separator, expressionList);
+        
+        final List<String> newExpressions;
+        if (Iterables.getLast(jexlNodeList) instanceof ASTERNode && !jexlNodeList.stream().allMatch(x -> x instanceof ASTERNode)) {
```

Comment:
This could be cleaned up by simply doing this when they are inserted and we encounter an er node. looking at the review I see that now

Reply:
> This could be cleaned up by simply doing this when they are inserted and we encounter an er node. looking at the review I see that now

Ha, I missed this comment. Yeah, that would probably be even better.

Reply:
So something like this for addComponent?
```java
    public void addComponent(JexlNode node) {
        Object lit = JexlASTHelper.getLiteralValue(node);
        String identifier = JexlASTHelper.getIdentifier(node);

        // If this is the first ASTERNode we're adding, then we want to escape
        // all of the previous expressions so that when the composite term is
        // regex expanded, the non-regex expressions don't accidentally get
        // counted as regexes (e.g., a normalized date has a '.' character in it).
        if ((node instanceof ASTERNode) && jexlNodeList.stream().noneMatch(n -> n instanceof ASTERNode)) {
            for (int i = 0; i < expressionList.size(); i++) {
                expressionList.set(i, Pattern.quote(expressionList.get(i)));
            }
        }

        jexlNodeList.add(node);
        fieldNameList.add(identifier);
        expressionList.add(lit.toString());
    }
```

Reply:
sure. why can't I accept this and just incorporate it? is that because it was submitted as a code block ?

Reply:
ha yeah, sometimes I write code then look at it in github and disagree with my initial approach. Or maybe I should say I "often" do that ;)

Reply:
> sure. why can't I accept this and just incorporate it? is that because it was submitted as a code block ?

Yeah, it's just a code block in a comment and not a suggestion. I think you can only make suggestions on changes already made in the PR. Since you didn't change anything in that part of the code, the UI doesn't give me a way to make a suggestion there.

Reply:
I guess my previous question still stands, though... Do we need to escape all non-regex terms (even those that occur after a regex term)? If so, then it might actually be better to do it where you had it before. I'm not sure, but I'd guess the answer depends on what is the regex. If it's something like `.*` then it probably doesn't matter. But if you had literal + regex + literal where the second literal had some regex chars in it and the actual regex was something like `foo|bar`, then you'd probably want to be sure the second literal was escaped too, wouldn't you?

Reply:
Hmmm, that's a good point. The external code protects that from happening because the rightmost term should be a regex, but there is no protection here. Perhaps That needs a little bit more thought.

Reply:
I need to add some unit tests to prove this. Will do that, then we can decide on whether to accept your change, below, or keep this change.